CMakeLists.txt

```diff
@@ -40,6 +40,7 @@ set(PROTO_BINARY_DIR ${CMAKE_BINARY_DIR}/build/generated)
 file(MAKE_DIRECTORY ${PROTO_BINARY_DIR})
 set(PROTO_IMPORT_DIRS proto/)
 
+#TODO: make it auto CPP detectable
```

Comment:
Apparently, according to the [CMake documentation](https://cmake.org/cmake/help/latest/command/file.html#glob), it is best practice to specify the list of source files manually. So we can actually remove this TODO again[^1] :)

[^1]: Although, this is actually incredible annoying and cumbersome in my opinion.

> **Note**
>
> We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild.